client/question/services/FeedbackGeneratorServiceSpec.js

```diff
@@ -171,61 +167,98 @@ describe('FeedbackGeneratorService', function() {
   });
 
   describe('_getCorrectnessTestFeedback', function() {
-    it([
-      'should return feedback on a user function that ',
-      'doesn\'t pass all correctness tests'
-    ].join(''), function() {
-      var testCase = TestCaseObjectFactory.create({
-        input: 'cat',
-        allowedOutputs: ['a']
-      });
-
-      var paragraphs = FeedbackGeneratorService
-        ._getCorrectnessTestFeedback(null, testCase, 'output').getParagraphs();
-      expect(paragraphs.length).toEqual(5);
-      expect(paragraphs[0].isTextParagraph()).toBe(true);
-      expect(paragraphs[0].getContent()).toBe(
-        "Your code produced the following result:");
-      expect(paragraphs[1].isCodeParagraph()).toBe(true);
-      expect(paragraphs[1].getContent()).toBe(
-        "Input: \"cat\"\nOutput: \"output\"");
-      expect(paragraphs[2].isTextParagraph()).toBe(true);
-      expect(paragraphs[2].getContent()).toBe(
-        "However, the expected output is:");
-      expect(paragraphs[3].isCodeParagraph()).toBe(true);
-      expect(paragraphs[3].getContent()).toBe("\"a\"");
-      expect(paragraphs[4].isTextParagraph()).toBe(true);
-      expect(paragraphs[4].getContent()).toBe(
-        "Could you fix this?");
+    var sampleInputTestSuite = {
+      id: 'SAMPLE_INPUT',
+      testCase: {
+        input: 'Hello, John',
+        allowedOutputs: ['olleH, nhoJ']
+      }};
+    var generalTestSuite = {
+      id: 'GENERAL_CASE',
+      testCase: {
+        input: 'Hi, world',
+        allowedOutputs: ['iH, dlrow']
+      }};
+    var sampleInputTestCase;
+    var generalInputTestCase;
+
+    beforeEach(function() {
+      sampleInputTestCase =
+        TestCaseObjectFactory.create(sampleInputTestSuite.testCase);
+      generalInputTestCase =
+        TestCaseObjectFactory.create(generalTestSuite.testCase);
     });
 
-    it([
-      'should return feedback with the output function name ',
-      'on a user function that doesn\'t pass all correctness tests'
-    ].join(''), function() {
-      var testCase = TestCaseObjectFactory.create({
-        input: 'cat',
-        allowedOutputs: ['a']
-      });
+    it('should allow user to display output if suite id is \'SAMPLE_INPUT\'',
+      function() {
+        var correctnessFeedbackParagraphs =
+          FeedbackGeneratorService._getCorrectnessTestFeedback(
+          sampleInputTestCase, 0, 'incorrect answer',
+          sampleInputTestSuite.id).getParagraphs();
+        var expectedCorrectnessFeedbackText =
```

Comment:
I'm afraid this isn't the right way to do this -- the expected/actual values should not need additional massaging, and setting "actual = expected" at the outset is a bit of a red flag. Here is how I would suggest writing the next 28 lines or so:

```
        var firstFeedbackParagraphText = correctnessFeedbackParagraphs[0].getContent();
        var possibleFeedbackTexts = CORRECTNESS_FEEDBACK_TEXT.OUTPUT_ENABLED;
        expect(possibleFeedbackTexts.includes(firstFeedbackParagraphText)).toEqual(true);

        expect(correctnessFeedbackParagraphs[0].isTextParagraph()).toBe(true);
        expect(correctnessFeedbackParagraphs[1].isOutputParagraph()).toBe(true);
        expect(correctnessFeedbackParagraphs.length).toEqual(2);
```

Similar feedback applies for other tests. Basically, don't declare proxy values and try to set them with complicated logic; instead, try to keep the tests as clean and straightforward as possible.

Reply:
The idea was to make it clear what the problem was. I would see vague failure messages saying the expected value of 5 was 2 or the expected value of true was false, making it very hard to look at the failure message and understand what exactly went wrong. That said, I'll change it as you suggested to make it consistent.